sklearn/impute/_iterative.py

```diff
@@ -581,6 +582,36 @@ def fit_transform(self, X, y=None):
             self.n_iter_ = 0
             return super()._concatenate_indicator(Xt, X_indicator)
 
+        # Convert min and max values to float arrays if array-like
+        # Else (i.e, if scalar) broadcast it to an array of shape=n_features
+        self._min_value = (
+            as_float_array(self.min_value, force_all_finite=False)
+            if _is_arraylike(self.min_value)
+            else as_float_array(np.full(X.shape[1], self.min_value),
+                                force_all_finite=False)
+        )
+        self._max_value = (
+            as_float_array(self.max_value, force_all_finite=False)
+            if _is_arraylike(self.max_value)
+            else as_float_array(np.full(X.shape[1], self.max_value),
+                                force_all_finite=False)
+        )
+        # Fill in any np.nan in min and max value arrays
+        # with -np.inf and np.inf respectively
+        self._min_value[np.isnan(self._min_value)] = -np.inf
```

Comment:
I would remove this case

Reply:
That case is removed, but keeping `self._min_value[np.isnan(self._min_value)] = -np.inf` (and same for max value) as that handles the case of `np.nan` and `None` in array inputs (not scalars) correctly. Look at the details of all the options I tried (and which didn't work) in my explanation below.

Reply:
As mentioned below we should not support None inside an array so they will be no issue